assets/src/models/createDetourMachine.ts

```diff
@@ -24,10 +73,120 @@ export const createDetourMachine = setup({
   initial: "Detour Drawing",
   states: {
     "Detour Drawing": {
-      initial: "Editing",
+      initial: "Begin",
       states: {
+        Begin: {
+          always: [
+            {
+              guard: ({ context }) =>
+                context.routePattern !== undefined &&
+                context.route !== undefined,
+              target: "Editing",
+            },
+            { target: "Pick Route Pattern" },
+          ],
+        },
+        "Pick Route Pattern": {
+          initial: "Pick Route ID",
```

Comment:
```suggestion
          initial: "Load Route Patterns",
```

Related to https://github.com/mbta/skate/pull/2652/files#r1630163656, I think we'll land in this state when the input has a route, but no route pattern, and when that happens, we want to be loading the route patterns.

(I had to make that change [here](https://github.com/mbta/skate/pull/2647/files#diff-2697d2574fc8e2dec397eba594a8b6fa60a8f51bf3ea7fa188e87d697a87bb50R90) in order to get the detour modal to load properly, but I think it makes more sense to make that change in this PR.

Reply:
UPDATE: ☝️ was based on an old version of this branch, and is no longer true. 

In fact, making my change would break this because the `Load Route Patterns` state doesn't exist anymore!